LayoutTests/imported/w3c/web-platform-tests/html/semantics/interactive-elements/the-details-element/toggleEvent.html

```diff
@@ -66,7 +66,7 @@
     assert_true(evt.isTrusted, "event is trusted");
     assert_false(evt.bubbles, "event doesn't bubble");
     assert_false(evt.cancelable, "event is not cancelable");
-    assert_equals(Object.getPrototypeOf(evt), Event.prototype, "Prototype of toggle event is Event.prototype");
+    assert_equals(Object.getPrototypeOf(evt), ToggleEvent.prototype, "Prototype of toggle event is ToggleEvent.prototype");
```

Comment:
Can we assert the oldState / newState property is set correctly every time as well?

Also, this will need to be exported to WPT (export-w3c-test-changes can help you with that)

Reply:
> Can we assert the oldState / newState property is set correctly every time as well?

Yep, I will add those

> Also, this will need to be exported to WPT (export-w3c-test-changes can help you with that)

aha, cool — will do (after review approval of the test updates here)

Reply:
I don't see the test updates here yet

Reply:
> I don't see the test updates here yet

Yeah, sorry, I’m still working on the test updates. (The most-recent force push was just to update/amend the commit message, not any code.)

Reply:
The newer revision still doesn't seem to have the assertions for old/newState

Reply:
> The newer revision still doesn't seem to have the assertions for old/newState

Right — that’s because when I added those assertions to the tests locally and ran them, the old/newState assertions I added to the two tests for _removing_ the `open` attribute are failing; specifically, my local WebKit build seems unexpectedly seems to be causing both the `newState` _and_ the `oldState` to be set to `closed` when the `open` attribute is removed.

So I guess I need to do more troubleshooting locally before I push the updated tests to this PR branch (because otherwise they’re just going to cause all the EWS bots to go red…)

Reply:
OK, in https://github.com/WebKit/WebKit/pull/16795/commits/5eb916675606f6af3bbbf5a981531ceaa8ab8729 I added the expected old/newState to all the _“adding open”_ tests.

However, I didn’t yet do the same for the _“removing open”_ tests, nor for the _“Calling open twice on 'details' fires only one toggle event”_ — because for those tests I’m not getting the expected states. Details:

- _“Removing open from 'details' should fire a toggle event at the 'details' element”_ (details2/t2)
  - expected: `oldState: open, newState: closed`
  - actual: `oldState: closed, newState: closed` ❌
- _“Calling removeAttribute('open') to 'details' should fire a toggle event at the 'details' element”_ (details8/t8)
  - expected: `oldState: open, newState: closed`
  - actual: `oldState: closed, newState: closed` ❌
  
_“Calling open twice on 'details' fires only one toggle event”_ also isn’t giving the expected results:
  - expected: `oldState: closed, newState: open`
  - actual: `oldState: closed, newState: closed` ❌

I’ll do some debugging, and post details from that in another comment.

Reply:
OK, in https://github.com/WebKit/WebKit/commit/84714b94de65e8425bea8c09b7091ebaa862d9b0 I added old/newState assertions for all tests, and fixed the code to produce the expected results.